### PR TITLE
feat: update approval check for self approval

### PR DESCRIPTION
**Pull request by utsav14nov**
Proton Pr: https://github.com/goto/proton/pull/123

### CODE REVIEW

**Comment by anjaliagg9791** [comment]
logic looks good to me.

**Review by Gadigeppa-J — APPROVED** [review]
lgtm

File: domain/policy.go [hunk 1]
```diff
@@ -310,6 +313,16 @@ func (p *Policy) RemoveSensitiveValues() {
 	}
 }
 
+func (p *Policy) GetStepByName(name string) *Step {
```
Comment (rahmatrhd) [inline domain/policy.go:316]: can you also add unit tests for this to help the test coverage?

**Review by rahmatrhd — CHANGES_REQUESTED** [review]

File: domain/policy.go [hunk 2]
```diff
@@ -107,6 +107,9 @@ type Step struct {
 	// Accessible parameters:
 	// $appeal = Appeal object
 	ApproveIf string `json:"approve_if,omitempty" yaml:"approve_if,omitempty" validate:"required_if=Strategy auto"`
+
+	// DontAllowSelfApproval is a boolean flag to detemine if the approver can approve their own request.
+	DontAllowSelfApproval bool `json:"dont_allow_self_approval,omitempty" yaml:"dont_allow_self_approval,omitempty"`
```
[the remark this answers is not in the document]
Comment (utsav14nov) [reply domain/policy.go:112]: @rahmatrhd The same was suggested on proton MR by @anjaliagg9791 .
Can you please check this https://github.com/goto/proton/pull/123#discussion_r1734006651

File: domain/policy.go [hunk 3]
```diff
@@ -310,6 +313,16 @@ func (p *Policy) RemoveSensitiveValues() {
 	}
 }
 
+func (p *Policy) GetStepByName(name string) *Step {
+	fmt.Printf("Policy: %v\n", p)
```
[the remark this answers is not in the document]
Comment (utsav14nov) [reply domain/policy.go]: Done

File: domain/policy.go (hunk 1 above)
Comment (utsav14nov) [reply domain/policy.go:316]: Done

**Review by rahmatrhd — APPROVED** [review]

**Pull request merged**